warehouse/forklift/legacy.py

```diff
@@ -1504,3 +1510,18 @@ def doc_upload(request):
         "Uploading documentation is no longer supported, we recommend using "
         "https://readthedocs.org/.",
     )
+
+
+@view_config(
+    route_name="forklift.legacy.redirect", require_csrf=False, require_methods=["POST"],
+)
+def permanent_redirect(request):
+    """
+    Deals with requests aimed at /legacy and points the user to the correct
+    /legacy/ route.
+    """
+    return _exc_with_message(
+        HTTPPermanentRedirect,
+        "It looks like you forgot a trailing /"
+        "Please make your POST request to /legacy/. This is /legacy",
```

Comment:
I think the POST bit may actually be a bit confusing to users seeing this, since I think the typical way someone will encounter this message will be by e.g. typoing the `--repository-url` in a call to `twine`, not by explicitly themselves making a POST request.

Maybe something like `An upload was attempted to /legacy, but the expected upload URL is /legacy/, with the trailing /.` avoids discussing the specifics of HTTP?

Reply:
+1 on the wording change